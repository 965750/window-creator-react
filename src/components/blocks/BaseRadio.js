import React from 'react'
import PropTypes from 'prop-types'
import styles from './BaseRadio.module.scss'

const BaseRadio = ({ isChecked, classes }) => (
    <div
        className={`${styles.radio__wrapper} border-botticelli border-4 rounded-full relative ${classes}`}
    >
        {isChecked ? (
            <div
                className={`${
                    styles['radio--checked']
                } top-half left-half rounded-full bg-hippieBlue absolute`}
            />
        ) : null}
    </div>
)

BaseRadio.propTypes = {
    isChecked: PropTypes.bool.isRequired,
    classes: PropTypes.string,
}

export default BaseRadio
