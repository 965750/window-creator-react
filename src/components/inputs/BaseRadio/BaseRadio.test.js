import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import BaseRadio from './index'
import messages from '../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('BaseRadio', () => {
  let wrapper
  const initialProps = {
    isChecked: false,
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <BaseRadio {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should pass classes from props to wrapper', () => {
    const mockClasses = 'mockClasses'

    setUp({
      ...initialProps,
      classes: mockClasses,
    })

    expect(wrapper.queryByTestId('radioWrapper')).toHaveClass(mockClasses)
  })

  it('should show inner dot when checked', () => {
    setUp({
      isChecked: true
    })

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
