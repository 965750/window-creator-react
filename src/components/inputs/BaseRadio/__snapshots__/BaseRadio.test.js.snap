// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseRadio should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="radio__wrapper border-botticelli border-4 rounded-full relative undefined"
    data-testid="radioWrapper"
  />
</DocumentFragment>
`;

exports[`BaseRadio should show inner dot when checked 1`] = `
<DocumentFragment>
  <div
    class="radio__wrapper border-botticelli border-4 rounded-full relative undefined"
    data-testid="radioWrapper"
  >
    <div
      class="radio--checked top-half left-half rounded-full bg-hippieBlue absolute"
    />
  </div>
</DocumentFragment>
`;
